iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -257,7 +258,8 @@ public Class<? extends AbstractSerDe> getSerDeClass() {
   public HiveMetaHook getMetaHook() {
     // Make sure to always return a new instance here, as HiveIcebergMetaHook might hold state relevant for the
     // operation.
-    return new HiveIcebergMetaHook(conf);
+    return StringUtils.isEmpty(MetaStoreUtils.getCatalogType(conf)) ?
```

Comment:
that won't work for Hive managed iceberg hadoop catalog. I think we need 2 type here: Hive managed and external catalogs.

Reply:
ATM hadoop catalog uses HiveIcebergMetaHook as well, however, I am not sure if it should